python/context.cpp

```diff
@@ -125,27 +125,15 @@ void register_contexts(pybind11::module& m) {
     // context
     pybind11::class_<context_shim, std::shared_ptr<context_shim>> context(m, "context", "An opaque handle for the hardware resources used in a simulation.");
     context
+        .def(pybind11::init([](){return make_context(arbenv::thread_concurrency(), pybind11::none(), pybind11::none());}),
+            "Construct a default context without arguments: defaults to the maximum number of threads the system makes available.")
         .def(pybind11::init(
             [](unsigned threads, pybind11::object gpu, pybind11::object mpi){
-                return create_context(threads, gpu, mpi);
+                return make_context(threads, gpu, mpi);
             }),
             "threads"_a=1, "gpu_id"_a=pybind11::none(), "mpi"_a=pybind11::none(),
             "Construct a distributed context with arguments:\n"
-            "  threads: The number of threads available locally for execution. Must be set to 1 at minimum. 1 by default.\n"
-            "  gpu_id:  The identifier of the GPU to use, None by default. Only available if arbor.__config__['gpu']==True.\n"
-            "  mpi:     The MPI communicator, None by default. Only available if arbor.__config__['mpi']==True.\n")
-        .def(pybind11::init(
-            [](std::string threads, pybind11::object gpu, pybind11::object mpi){
-                if ("avail_threads" == threads) {
-                    return create_context(arbenv::thread_concurrency(), gpu, mpi);
-                }
-                throw pyarb_error(
-                        util::pprintf("Attempt to set threads to {}. The only valid thread options are a positive integer greater than 0, or 'avial_threads'.", threads));
-
-            }),
-            "threads"_a, "gpu_id"_a=pybind11::none(), "mpi"_a=pybind11::none(),
-            "Construct a distributed context with arguments:\n"
-            "  threads: A string option describing the number of threads. Currently, only \"avail_threads\" is supported.\n"
+            "  threads: The number of threads available locally for execution. Must be set to 1 at minimum. Defaults to the maximum number of threads the system makes available if gpu_id and mpi are not set, else defaults to 1.\n"
```

Comment:
```suggestion
            "  threads: The number of threads to be used. Must be set to 1 at minimum. Defaults to the maximum number of threads the system makes available if gpu_id and mpi are not set, else defaults to 1.\n"
```
This argument is prescriptive rather than descriptive. Arbor will not decide to use less threads than the number you pass here.
Also, technically, it's not only execution, but also construction.